83 client cfg push

RmbRT opened the pull request:
Closes #83

sebastianst commented on config/client.go:
```diff
@@ -18,6 +18,8 @@ func ParseClientConfig() (cfg ClientConfig) {
 	flag.StringVar(&cfg.ChainURL, "chain-url", "ws://127.0.0.1:8545", "'protocol://ip:port'")
 	flag.StringVar(&cfg.OpHost, "op-host", "127.0.0.1", "IP of hostname")
 	flag.IntVar(&cfg.OpPort, "op-port", 8401, "Account derivation index.")
+	flag.StringVar(&cfg.OpHost, "op-host", "127.0.0.1", "IP/host name of operator")
+	flag.IntVar(&cfg.OpPort, "op-port", 8401, "Port of operator.")
```
The first commit adds there two duplicate lines but doesn't remove the old ones.

sebastianst commented on client/rpc.go:
```diff
@@ -152,23 +165,33 @@ func (r *RPC) handleConnections() {
+			}
+			close(r.hasConfig)
+			hasConfig = true
+		} else {
```
I think it's easier to read if you add a `continue` in the first `if` branch and then put the normal path of execution after the `if`, instead of having it inside the `else` branch, which is always executed except for the very first msg.

sebastianst commented on gui/gui.go:
```diff
@@ -187,7 +187,8 @@ func layout(g *gocui.Gui, client *client.Client) error {
 		addr := client.Address().Hex()[:10]
-		v.Title = client.Config.UserName + fmt.Sprintf(" %s on %s", addr, client.Config.ChainURL)
+		v.Title = client.Config.UserName + fmt.Sprintf(" %s on %s", addr,
+			client.Config.ChainURL(client.OpConfig.NetworkID))
```
I think this expression is used several times and is worth being put into a method like `client.ChainURL() string`.

RmbRT committed:
config: Added chain URL dictionary to client cfg.

Signed-off-by: Steffen Rattay <steffen@perun.network>

RmbRT committed:
config: Fixed config description.

Signed-off-by: Steffen Rattay <steffen@perun.network>

RmbRT committed:
client: Handled client config in RPC.

Signed-off-by: Steffen Rattay <steffen@perun.network>

RmbRT committed:
gui: Adapted to new config architecture.

Signed-off-by: Steffen Rattay <steffen@perun.network>

RmbRT committed:
client: Adapted integration test to new config.

Signed-off-by: Steffen Rattay <steffen@perun.network>

RmbRT committed:
client: switched to use operator's client config.

- Removed client's contract address config option.
- Client now constructed with operator's ClientConfig.

Signed-off-by: Steffen Rattay <steffen@perun.network>
